patches/minecraft/net/minecraft/world/gen/feature/jigsaw/JigsawPiece.java.patch

```diff
@@ -0,0 +1,18 @@
+--- a/net/minecraft/world/gen/feature/jigsaw/JigsawPiece.java
++++ b/net/minecraft/world/gen/feature/jigsaw/JigsawPiece.java
+@@ -5,6 +5,7 @@
+ import java.util.List;
+ import java.util.Random;
+ import javax.annotation.Nullable;
++import net.minecraft.util.ResourceLocation;
+ import net.minecraft.util.Rotation;
+ import net.minecraft.util.math.BlockPos;
+ import net.minecraft.util.math.MutableBoundingBox;
+@@ -61,4 +62,7 @@
+    public int func_214850_d() {
+       return 1;
+    }
++
++   public abstract ResourceLocation getIdentifier();
```

Comment:
Adding abstract methods is a binary break, as other mod systems wont implement this and get abstract method exception, so you need to find a sane default.